src/Lachain.Consensus/Messages/MessageEnvelopeRepositoryManager.cs

```diff
@@ -0,0 +1,79 @@
+using System;
+using System.Collections.Generic;
+using Lachain.Logger;
+using Lachain.Storage.Repositories;
+
+namespace Lachain.Consensus.Messages
+{
+    public class MessageEnvelopeRepositoryManager
+    {
+        private readonly IMessageEnvelopeRepository _repository;
+        private MessageEnvelopeList? _messageEnvelopeList;
+        private static readonly ILogger<MessageEnvelopeRepositoryManager> logger = LoggerFactory.GetLoggerForClass<MessageEnvelopeRepositoryManager>();
+
+        public bool IsPresent => !(_messageEnvelopeList is null);
+        public MessageEnvelopeRepositoryManager(IMessageEnvelopeRepository repository)
+        {
+            _repository = repository;
+        }
+
+        public void LoadFromDb()
+        {
+            var bytes = _repository.LoadMessages();
+            _messageEnvelopeList = !(bytes is null) ? MessageEnvelopeList.FromByteArray(bytes) : null;
+        }
+
+        public long GetEra()
+        {
+            if (!IsPresent)
+            {
+                throw new InvalidOperationException("Could not find MessageEnvelopeList in repository");
+            }
+            return _messageEnvelopeList.Era;
+        }
+
+        public void StartEra(long era, bool canBeSame = false)
+        {
+            if (!canBeSame && IsPresent && _messageEnvelopeList.Era == era)
+            {
+                throw new ArgumentException($"Start Era called with same era number {era}");
+            }
+     
+            _messageEnvelopeList = new MessageEnvelopeList(era);
+            SaveToDb(_messageEnvelopeList);
+        }
+
+        public void AddMessage(MessageEnvelope message)
+        {
+            if (!IsPresent)
+            {
+                throw new InvalidOperationException("Could not find MessageEnvelopeList in db");
+            }
+
+            try
+            {
+                _messageEnvelopeList.AddMessage(message);
+                SaveToDb(_messageEnvelopeList);
```

Comment:
Instead of saving the whole list to db, save one message at a time. Saving whole list is too costly if some malicious validator spams msgs. And also implement msg deletion once consnensus is terminated for current era